apps/shell/tash_command.c

```diff
@@ -564,22 +567,27 @@ int tash_execute_cmd(char **args, int argc)
 				pmdrv_fd = open("/dev/pm", O_WRONLY);
 				if (pmdrv_fd < 0) {
 					shdbg("open /dev/pm failed(%d), \n", get_errno());
+					ret = ERROR;
 				} else {
-					pm_suspended = ioctl(pmdrv_fd, PMIOC_SUSPEND, PM_NORMAL);
-					if (pm_suspended != OK) {
+					if (ioctl(pmdrv_fd, PMIOC_DOMAIN_REGISTER, &pm_domain_arg) != OK) {
+						shdbg("pm_domain_register failed(%d)\n", get_errno());
+						ret = ERROR;
+					} else if (ioctl(pmdrv_fd, PMIOC_SUSPEND, pm_domain_arg.domain_id) != OK) {
 						shdbg("pm_suspend failed(%d)\n", get_errno());
+						ret = ERROR;
 					}
 				}
 #endif
 				(*tash_cmds_info.cmd[cmd_idx].cb) (argc, args);
 #ifdef CONFIG_PM
-				if (pmdrv_fd > 0) {
-					if (pm_suspended == OK) {
-						/* pm_resume should only be called when the pm_suspend is executed correctly. */
-						if (ioctl(pmdrv_fd, PMIOC_RESUME, PM_NORMAL) != OK) {
-							shdbg("pm_resume failed(%d)\n", get_errno());
-						}
+				if (ret == OK) {
```

Comment:
you don't return using the ret.
so, how to keep this checking value name to pm_suspended